Signal/src/ViewControllers/MessageMetadataViewController.swift

```diff
@@ -147,47 +160,108 @@ class MessageMetadataViewController: OWSViewController {
+
+                        addDivider()
+                    }
+
+                    let cell = ContactTableViewCell()
+                    cell.configure(withRecipientId: recipientId, contactsManager: self.contactsManager)
+                    let statusLabel = UILabel()
+                    statusLabel.text = statusMessage
+                    statusLabel.textColor = UIColor.ows_darkGray()
+                    statusLabel.font = UIFont.ows_footnote()
+                    statusLabel.sizeToFit()
+                    cell.accessoryView = statusLabel
+                    cell.autoSetDimension(.height, toSize:ContactTableViewCell.rowHeight())
+                    cell.setContentHuggingLow()
+                    groupRows.append(cell)
+                }
+
+                if groupRows.count > 0 {
+                    addDivider()
+
+                    let spacer = UIView()
+                    spacer.autoSetDimension(.height, toSize:10)
+                    groupRows.append(spacer)
+                }
+
+                Logger.verbose("\(groupRows.count) rows for \(recipientStatusGroup)")
+                guard groupRows.count > 0 else {
+                    continue
+                }
+                rows += groupRows
             }
         }
 
         rows.append(valueRow(name: NSLocalizedString("MESSAGE_METADATA_VIEW_SENT_DATE_TIME",
                                                      comment: "Label for the 'sent date & time' field of the 'message metadata' view."),
                              value:DateUtil.formatPastTimestampRelativeToNow(message.timestamp)))
 
-        if let _ = message as? TSIncomingMessage {
+        if message as? TSIncomingMessage != nil {
             rows.append(valueRow(name: NSLocalizedString("MESSAGE_METADATA_VIEW_RECEIVED_DATE_TIME",
                                                          comment: "Label for the 'received date & time' field of the 'message metadata' view."),
                                  value:DateUtil.formatPastTimestampRelativeToNow(message.timestampForSorting())))
         }
 
+        rows += addAttachmentMetadataRows()
+
         // TODO: We could include the "disappearing messages" state here.
 
+        var lastRow: UIView?
+        for row in rows {
+            contentView.addSubview(row)
+            row.autoPinLeadingToSuperView()
+            row.autoPinTrailingToSuperView()
+
+            if let lastRow = lastRow {
+                row.autoPinEdge(.top, to:.bottom, of:lastRow, withOffset:5)
+            } else {
+                row.autoPinEdge(toSuperviewEdge:.top, withInset:20)
+            }
+
+            lastRow = row
+        }
+        if let lastRow = lastRow {
+            lastRow.autoPinEdge(toSuperviewEdge:.bottom, withInset:20)
+        }
+
+        if let mediaMessageView = mediaMessageView {
+            mediaMessageView.autoPinToSquareAspectRatio()
+        }
+    }
+
+    private func addContentRows() -> [UIView] {
```

Comment:
nit: `addContentRows` reads like a mutator. Should this be just `contentRows()`?

Reply:
Done.